walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -39,30 +39,67 @@
                 .map(e -> e.getWhen())
                 .collect(Collectors.toList());
         Collections.sort(eventTimeRanges, TimeRange.ORDER_BY_START);
-        
-        // Check for invalid input of duration.
-        if(duration > TimeRange.WHOLE_DAY.duration()) {
-            return validTimeRanges;
+
+        // First get the time ranges that work for all attendees. 
+        List<TimeRange> eventTimeRangesWithAllAttendees = 
+            getValidEventTimeRanges(events, request.getAllAttendees());
+        Collection<TimeRange> meetingTimeRangesWithAllAttendees = 
+            getMeetingTimeRanges(eventTimeRangesWithAllAttendees, duration);
+
+        // If there are no time ranges that work for all attendees, get the time
+        // ranges that work for only the mandatory attendees.
+        if (meetingTimeRangesWithAllAttendees.size() == 0) {
+            List<TimeRange> eventTimeRangesWithMandatoryAttendees = 
+                getValidEventTimeRanges(events, request.getAttendees());
+            
+            // In the case that there are optional attendees but no mandatory attendees, 
+            // treat the optional attendees as the mandatory attendees
+            if (eventTimeRangesWithMandatoryAttendees.size() == 0 && eventTimeRangesWithAllAttendees.size() != 0 ){
```

Comment:
The comment here doesn't match the code. Even if they're semantically equivalent, they should match.

Consider changing to if (request.getAttendees().size() == 0) { return meetingTimeRangesWithAllAttendees; }

You could even pull it out of this if block for more clarity

Reply:
Ah okay. Thank you!